### PR TITLE
Move CLI config processing out from under registry dir

### PR DESCRIPTION
**Pull request by duglin**
No logic changes should be in here, just moving things around.

Signed-off-by: Doug Davis dug@us.ibm.com

### CODE REVIEW

**Comment by jfrazelle** [comment]
should it be top level or pkg/cliconfig, nm i guess runconfig is top level, dont want to start a bike shed haha

seems sane to me

LGTM

**Comment by duglin** [comment]
ping @dmcgowan

File: cliconfig/config.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+	"strings"
+
+	"github.com/docker/docker/pkg/homedir"
+)
+
+const (
+	// Where we store the config file
+	CONFIGFILE     = "config.json"
+	OLD_CONFIGFILE = ".dockercfg"
+	INDEXSERVER    = "https://index.docker.io/v1/"
```
Comment (dmcgowan) [inline cliconfig/config.go]: I would prefer to keep these variables inside the registry package.  They will likely disappear with the registry refactor, which I don't think should require changes here. Same goes for the functions which use these `IndexServerAddress()` and `IndexServerName()`

Comment (duglin) [inline cliconfig/config.go]: I would too but I couldn't. :-(  I got into an annoying circular dependency.  If I move these back into registry then cliconfig imports registry and then registry needs to import cliconfig for the config stuff.  
Open to ideas on how to fix it though.

File: cliconfig/config.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+		authConfig.Username, authConfig.Password, err = DecodeAuth(origAuth[1])
+		if err != nil {
+			return &configFile, err
+		}
+		origEmail := strings.Split(arr[1], " = ")
+		if len(origEmail) != 2 {
+			return &configFile, fmt.Errorf("Invalid Auth config file")
+		}
+		authConfig.Email = origEmail[1]
+		authConfig.ServerAddress = IndexServerAddress()
```
Comment (dmcgowan) [inline cliconfig/config.go]: This value can be passed in as the default server, then callers would take care of defining the default at registry.IndexServerAddress()

**Comment by duglin** [comment]
@dmcgowan ok - moved that stuff back!  thanks for the help!

**Comment by dmcgowan** [comment]
LGTM

Also +1 to @jfrazelle, there is likely a better package name, but if we don't find one I don't think that should block getting merged.

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**